### DISCUSSION TITLE
Where are the docs for "Not writable" in thumbnail settings?

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#11008] [Q&A / Support]
Reading through https://pimcore.com/docs/pimcore/current/Development_Documentation/Assets/Working_with_Thumbnails/Image_Thumbnails.html I cannot find any details why some thumbnail configurations are writeable and some are not. Where is it?

![image](https://user-images.githubusercontent.com/279826/145837454-3cf00d00-eeea-4568-bd1d-466d97c5802f.png)

**Comment by dvesh3** [comment]
please see https://pimcore.com/docs/pimcore/current/Development_Documentation/Deployment/Configuration_Environments.html#page_Production-environment-with

**Reply by NiklasBr** [reply]
Thanks, that was certainly unclear from everything about the error message.

**Reply by NiklasBr** [reply]
@dvesh3 Because Image thumbnails now appears to be locked by default, what is the expected way to invalidate a single thumbnail?

**Reply by dvesh3** [reply]
Do you mean existing thumbnails? you have to revalidate them in dev environment(with debug mode on) and then deploy it on production.

**Reply by NiklasBr** [reply]
If I check in changes and deploy them I expect that the target environment be able to revalidate the current or new config. Previously this validation led to invalid thumbnails being re-generated with the new settings, this appears to no longer happen, and it is exacerbated by the new lock which appears to have been recently introduced.

**Reply by dvesh3** [reply]
Also was not the case with old php config files, so you have to resave the thumbnail config on production to invalidate thumbnails. (which is not possible anymore since the save button is disabled). I agree, there should be some way to invalidate thumbnail configs, so a command would be nice to invalidate thumbnails based on config. In the command, we just need to call `clearTempFiles()` on thumbnail config.

Would you like to contribute a PR?

**Comment by pdchaudhary** [comment]
```
PIMCORE_WRITE_TARGET_IMAGE_THUMBNAILS=settings-store
PIMCORE_WRITE_TARGET_CUSTOM_REPORTS=settings-store
PIMCORE_WRITE_TARGET_VIDEO_THUMBNAILS=settings-store
PIMCORE_WRITE_TARGET_DOCUMENT_TYPES=settings-store
PIMCORE_WRITE_TARGET_WEB_TO_PRINT=settings-store
PIMCORE_WRITE_TARGET_PREDEFINED_PROPERTIES=settings-store
PIMCORE_WRITE_TARGET_PREDEFINED_ASSET_METADATA=settings-store
PIMCORE_WRITE_TARGET_STATICROUTES=settings-store
PIMCORE_WRITE_TARGET_PERSPECTIVES=settings-store
PIMCORE_WRITE_TARGET_CUSTOM_VIEWS=settings-store
```

@NiklasBr Add above code in .env file
